core/src/test/java/org/elasticsearch/index/query/QueryFilterBuilderTest.java

```diff
@@ -50,7 +54,7 @@ protected QueryFilterBuilder createTestQueryBuilder() {
      */
     @Test
     public void testNoInnerQuery() throws QueryParsingException, IOException {
-        QueryFilterBuilder queryFilterQuery = new QueryFilterBuilder(null);
+        QueryFilterBuilder queryFilterQuery = new QueryFilterBuilder(EmptyQueryBuilder.PROTOTYPE);
         assertNull(queryFilterQuery.toQuery(createContext()));
     }
 
```

Comment:
Maybe we should also add small tests for all of the queries where we added null checks in constructors?